### DISCUSSION TITLE
How can i update all the relation data ?

### DISCUSSION THREAD

**Discussion by myNameIsDu** [prisma/prisma#14374] [Q&A] [answered]
When I use `updateMany`, it told me that  I must have a `where`. And then I gave it an empty object. it works.

<img width="338" alt="image" src="https://user-images.githubusercontent.com/53305259/179685786-a45c7f01-3b2a-47fc-a5c3-92891a11a6e9.png">

Maybe there's a better way

**Comment by austincrim** [comment, marked as answer]
Hey @myNameIsDu 👋, this is the correct way of accomplishing this, no need to change it.

**Reply by myNameIsDu** [reply]
Thanks so much for answering my question. But there's still a question.

<img width="442" alt="image" src="https://user-images.githubusercontent.com/53305259/179903736-38b93ccf-969b-4365-97e4-eb5ccb6a5712.png">

If I want to use `nested update` in  `updateMany`, It gave me an error. I checked the type file, and I found it without relation type.
`updateMany` doesn't support `nested update`?
So, I had to execute the query for each one and then execute the update for each one.

<img width="665" alt="image" src="https://user-images.githubusercontent.com/53305259/179905289-e202b166-2a66-48b6-b9c7-e98a76799cb9.png">

Is there a better way to do this? Or this is the best way?

Thank you for help or advice.

**Reply by myNameIsDu** [reply]
I found out it in the [document](https://www.prisma.io/docs/guides/performance-and-optimization/prisma-client-transactions-guide#nested-writes):

<img width="683" alt="image" src="https://user-images.githubusercontent.com/53305259/179915578-5eb8ac65-b993-4b72-8f4c-cfd299168560.png">

But I'm still curious why it doesn't support it, It's an easy operation.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
